roles/cnf_cert/tasks/tests.yml

```
@@ -11,37 +23,96 @@
   when: "'preflight' in tnf_labels and partner_creds | length > 0"
   ansible.builtin.include_tasks: prepare_json_authentication.yml
 
+- name: Copy kubeconfig file in config folder
+  ansible.builtin.copy:
+    src: "{{ kubeconfig_path }}"
+    dest: "{{ tnf_dir }}/config_files/kubeconfig"
+    mode: '0640'
+
 - name: Remove XML JUnit results if they exist
   ansible.builtin.file:
-    path: "{{ tnf_dir }}/{{ test_network_function_project_name }}/cnf-certification-tests_junit.xml"
+    path: "{{ tnf_dir }}/result_files/cnf-certification-tests_junit.xml"
     state: absent
 
+# This way of launching CNF Cert Suite is deprecated since tnf v5.1.1.
 # The task automatically append these environment variables to tnf_env_vars:
 # - TNF_PARTNER_REPO
 # - TNF_IMAGE
 # - SUPPORT_IMAGE
-# We have to set ignore_errors to true because, if any tnf test fails, the
-# result of the execution will be failed.  We will check log files afterwards
-# to really determine if there were errors in the execution or not.
-- name: Run the CNF Test Suite
+# We have to set ignore_errors to true (for tnf < v5.0.4) because, if any tnf test
+# fails, the result of the execution will be failed. We will check log files
+# afterwards to really determine if there were errors in the execution or not.
+- name: Run the CNF Test Suite (tnf <= v5.1.0)
+  when:
+    - test_network_function_version is version("v5.1.0", "<=")
+    - test_network_function_version != "HEAD"
   vars:
     tnf_partner_repo: "{{ (dci_local_registry | length) | ternary(dci_local_registry, 'quay.io') }}/testnetworkfunction"
     support_image: "debug-partner:{{ support_image_version }}"
-    tnf_exec_ignore_errors: "{{ (test_network_function_version is version('v5.0.4', '>=') or test_network_function_version == 'HEAD') | ternary('false', 'true') }}"
+    tnf_exec_ignore_errors: "{{ (test_network_function_version is version('v5.0.4', '>=')) | ternary('false', 'true') }}"
   environment: "{{ tnf_env_vars | combine({'TNF_PARTNER_REPO': tnf_partner_repo, 'TNF_IMAGE': tnf_image, 'SUPPORT_IMAGE': support_image}) }}"
   ansible.builtin.shell: |
     set -x
 
-    ./run-tnf-container.sh -k {{ kubeconfig_path }} \
-    -t {{ tnf_dir }}/{{ test_network_function_project_name }} \
-    -o {{ tnf_dir }}/{{ test_network_function_project_name }} \
+    ./run-tnf-container.sh -k {{ tnf_dir }}/config_files/kubeconfig \
```

Maybe like this since it's already inside `tnf_dir` ?
```suggestion
    ./run-tnf-container.sh -k config_files/kubeconfig \
```

---

I think I've been passing the arguments using the absolute path because I had issues with relative paths in the past. But I'll test this.

---

As commented in https://github.com/redhatci/ansible-collection-redhatci-ocp/pull/331#issuecomment-2149295177, we cannot use relative paths for the arguments passed to `run-tnf-container`, so I will keep the code in the same status if you're fine with this.
I'll also keep `chdir: "{{ tnf_dir }}"` just to shortcut the call to `./run-tnf-container.sh` script, which is placed in that directory